syllabus/core/dual_curriculum_wrapper.py

```diff
@@ -0,0 +1,60 @@
+from typing import Tuple, TypeVar
+
+from syllabus.core import (  # noqa: E402
+    Curriculum,
+    TaskWrapper,
+    make_multiprocessing_curriculum,
+)
+
+AgentID = TypeVar("AgentID")
+Agent = TypeVar("Agent")
+EnvTask = TypeVar("EnvTask")
+AgentTask = TypeVar("AgentTask")
+
+
+class DualCurriculumWrapper:
+    """Curriculum wrapper containing both an agent and environment-based curriculum."""
+
+    def __init__(
+        self,
+        env: TaskWrapper,
+        env_curriculum: Curriculum,
+        agent_curriculum: Curriculum,
+        *args,
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.env = env
+        self.agent_curriculum = agent_curriculum
+        self.env_curriculum = env_curriculum
+
+        self.env_mp_curriculum, self.env_task_queue, self.env_update_queue = (
+            make_multiprocessing_curriculum(env_curriculum)
+        )
+        self.agent_mp_curriculum, self.agent_task_queue, self.agent_update_queue = (
+            make_multiprocessing_curriculum(agent_curriculum)
+        )
+        self.sample()  # initializes env_task and agent_task
```

Comment:
This initializer seems hacky. I don't think we should be internally making multiprocessing curricula. Instead let the user create 2 curricula, pass them to this, then in their code, wrap the dual curriculum in a multiprocessing wrapper. This wrapper should implement update functions to pass updates to both the agent and environment curriculum. For example the update_on_step function should call self.env_curriculum.update_on_step and self.agent_curriculum.update_on_step.

I also don't see why we need to call sample here?

Reply:
The initializer might also need to update the task space. So the task space of this curriculum should be `Tuple(env_curriculum.task_space, agent_curriculum.task_space)`

I'll try to merge Nistha's task space updates in today so that you can pull them in here